### PR TITLE
Group assessment questions

### LINKED ISSUE #3869: Group assessments questions
Ability to group questions of assessments together.

The ability to group questions of assessments together would provide the following benefits:
- It prevents unnecessary repetition in questions, which are sharing the same topic.
- It enables grouping of results in a similar fashion, which helps to understand. If questions sets (groups) are switched, it could be used to help to understand in which context the question has been answerd.

Additional requirements:
- Each group / set must allow a label grouping the questions
- The results must be displayed accordingly
- Provide the question group / set with it's own `test` property, to prevent unnecessary repition of test criteria amongst the questions

### PR DESCRIPTION
**Pull request by anilsonmez-simsoft**
Assessment questions can be grouped through the dictionary using the `questionSets` property.

Closes #3869
Closes [AB#216](https://dev.azure.com/ncia-anet/2aa083a5-af3d-44e1-8c7b-6e9e6b124d91/_workitems/edit/216)

#### User changes
- None

#### Super User changes
- None

#### Admin changes
- None

#### System admin changes
- [x] anet.yml or anet-dictionary.yml needs change:
  you can configure `questionSets` properties for the assessments in the dictionary to group assessment questions
- [ ] db needs migration
- [ ] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [x] Described the user behavior in PR body
  - [x] Referenced/updated all related issues
  - [x] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [x] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [ ] Added and/or updated unit tests
  - [ ] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [ ] Updated documentation
  - [x] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: client/src/components/assessments/OndemandAssessments.js [hunk 1]
```diff
@@ -227,14 +230,27 @@ const OnDemandAssessments = ({
...
+                      <>
+                        <ReadonlyCustomFields
+                          parentFieldName={parentFieldName}
+                          fieldsConfig={filteredAssessmentConfig.questions}
+                          values={{
+                            [parentFieldName]: assessmentFieldsObject
+                          }}
+                          vertical
+                        />
+                        <QuestionSet
```
Comment (asafkaganbezgin) [inline client/src/components/assessments/OndemandAssessments.js:242]: Should this be conditionally rendered, or is this approach required?

File: client/src/components/assessments/PeriodicAssessmentResults.js [hunk 2]
```diff
@@ -116,12 +118,22 @@ const PeriodicAssessment = ({
...
-                  vertical
-                />
+                <>
+                  <ReadonlyCustomFields
+                    parentFieldName={parentFieldName}
+                    fieldsConfig={assessmentConfig.questions}
+                    values={values}
+                    vertical
+                  />
+                  <QuestionSet
```
Comment (asafkaganbezgin) [inline client/src/components/assessments/PeriodicAssessmentResults.js:128]: Should this be conditionally rendered, or is this approach required?

File: client/tests/webdriver/baseSpecs/assessmentsForPeople.spec.js [hunk 3]
```diff
@@ -28,13 +35,22 @@ describe("For the periodic person assessments", () => {
...
 
       ShowPerson.addPeriodicAssessmentButton.click()
       ShowPerson.waitForAssessmentModalForm()
+      ShowPerson.topLevelQuestionSetTitle.waitForDisplayed()
+      ShowPerson.bottomLevelQuestionSetTitle.waitForDisplayed()
+    })
 
+    it("Should display validation error messages on every level", () => {
+      ShowPerson.saveAssessmentButton.click()
+      expect(ShowPerson.getValidationErrorMessages()).to.eql(ERROR_MESSAGES)
```
Comment (asafkaganbezgin) [inline client/tests/webdriver/baseSpecs/assessmentsForPeople.spec.js:50]: `to.equal()` can be used to be consistent.

Comment (anilsonmez-simsoft) [reply client/tests/webdriver/baseSpecs/assessmentsForPeople.spec.js:50]: They are different actually. `to.equal()` cannot be used here as it checks if they are strictly equal while `to.eql() `checks deep equality.

File: client/src/components/assessments/InstantAssessmentResults.js [hunk 4]
```diff
@@ -63,3 +63,85 @@ InstantAssessmentsRow.propTypes = {
   questionConfig: PropTypes.object.isRequired,
```
Comment (asafkaganbezgin) [inline client/src/components/assessments/InstantAssessmentResults.js:63]: Sometimes, person pages give the following warning -> Failed prop type: Invalid prop `questionConfig` of type `boolean` supplied to `InstantAssessmentsRow`, expected `object`. 
It is always shown if you navigate to a person show page from the person links inside process board. However, it also appears from time to time when you navigate from different links on different pages. Very hard to reproduce those cases.

**Review by asafkaganbezgin — APPROVED** [review]

**Pull request merged**